### PR TITLE
Change the if condition to check if build is installed

### PR DESCRIPTION
**Pull request by AnandInguva**
Fixes: https://github.com/apache/beam/issues/28990
Fixes: https://github.com/apache/beam/issues/29020

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/runners/portability/stager.py [hunk 1]
```diff
@@ -774,7 +774,7 @@ def _build_setup_package(setup_file,  # type: str
       if build_setup_args is None:
         # if build is installed in the user env, use it to
         # build the sdist else fallback to legacy setup.py sdist call.
-        if importlib.util.find_spec('build'):
+        if pkgutil.find_loader('build') is not None:
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/portability/stager.py:777]: this is a deprecated method that will be removed in a future version of Python.

Comment (AnandInguva) [reply sdks/python/apache_beam/runners/portability/stager.py]: Using try except instead.

**Comment by AnandInguva** [comment]
R: @damccorm @riteshghorse @jrmccluskey since Valentyn is on leave

**Review by jrmccluskey — APPROVED** [review]
LGTM

**Comment by AnandInguva** [comment]
Run Python 3.8 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.9 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.10 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.11 PostCommit

**Comment by jrmccluskey** [comment]
Run Python_Coverage PreCommit

**Comment by jrmccluskey** [comment]
Run Python_Runners PreCommit

**Comment by AnandInguva** [comment]
PostCommits are passing. Merging

**Pull request merged**